### PR TITLE
Container metrics

### PR DESCRIPTION
**Pull request by aermakov-zalando**
Create entities for containers, not just pods.

### CODE REVIEW

**Comment by mohabusama** [comment]
I was just wondering about the duplication here. Almost all the info exists in the pod entity and we just duplicate in a new entity.

Is the reason for adding separate entities is to filter on ``type:kube_pod_container``? Is this not achievable more or less via ``type:kube_pod``?

I was also thinking that we can just merge the container entities in the same function of ``get_cluster_pods``; since we are just doing the loop twice with redundant info. So we can just rename to ``get_cluster_pods_and_containers(...)``

**Comment by aermakov-zalando** [comment]
As discussed: I want to track charts and build dashboards and alerts that would track metrics on individual containers. Instead of doing it all manually or semi-manually, I propose we add containers as a top-level entity in ZMON and start creating them in zmon-agent-core.

I've changed the implementation to build the list in one go (and later filter in `get_cluster_nodes`).

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 1]
```diff
@@ -188,27 +191,50 @@ def get_cluster_pods(kube_client, cluster_id, alias, environment, region, infras
...
+                'image': container_image,
+                'ready': container_ready,
+                'restarts': container_restarts,
+                'ports': container_ports
+            }
 
-    return entities
+            container_entity.update(pod_labels)
+            container_entity.update(shared_properties)
+            yield container_entity
```
Comment (mohabusama) [inline zmon_agent/discovery/kubernetes/cluster.py:234]: I suggest we keep it consistent with our returns. Either ``yield`` on all functions or just return a list.

Comment (aermakov-zalando) [reply zmon_agent/discovery/kubernetes/cluster.py:234]: I'd change it to `yield` everywhere, then. What do you think?

**Comment by mohabusama** [comment]
👍

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**